tools/migration/migrations/repo-1-2/migration.go

```diff
@@ -1 +1,380 @@
-package repo12
+package migration12
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/repo"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// duplicate head key here to protect against future changes
+var headKey = datastore.NewKey("/chain/heaviestTipSet")
+
+// migrationChainStore is a stripped down implementation of the Store interface
+// based on Store, containing only the fields and functions needed for the migration.
+//
+// the extraction line is drawn at package level where the migration occurs, i.e. chain,
+// and no further.
+type migrationChainStore struct {
+	// BsPriv is the on disk storage for blocks.
+	BsPriv bstore.Blockstore
+
+	// Ds is the datastore backing bsPriv.  It is also accessed directly
+	// to set and get chain meta-data, specifically the tipset cidset to
+	// state root mapping, and the heaviest tipset cids.
+	Ds repo.Datastore
+
+	// head is the tipset at the head of the best known chain.
+	Head types.TipSet
+}
+
+// GetBlock retrieves a block by cid.
+func (store *migrationChainStore) GetBlock(ctx context.Context, c cid.Cid) (*types.Block, error) {
+	data, err := store.BsPriv.Get(c)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get block %s", c.String())
+	}
+	return types.DecodeBlock(data.RawData())
+}
+
+// MetadataFormatJSONtoCBOR is the migration from version 1 to 2.
+type MetadataFormatJSONtoCBOR struct {
+	store *migrationChainStore
+}
+
+// Describe describes the steps this migration will take.
+func (m *MetadataFormatJSONtoCBOR) Describe() string {
+	return `MetadataFormatJSONtoCBOR migrates the storage repo from version 1 to 2.
+
+    This migration changes the chain store metadata serialization from JSON to CBOR.
+    The chain store metadata will be read in as JSON and rewritten as CBOR. 
+	Chain store metadata consists of associations between tipset keys and state 
+	root cids and the tipset key of the head of the chain.
+	No other repo data is changed.  Migrations are performed on a copy of the
+	chain store.
```

Comment:
This last statement is not under the control of this migration, so I suggest dropping it